WordPress/src/test/java/org/wordpress/android/ui/reader/subfilter/SubFilterViewModelTest.kt

```diff
@@ -238,11 +288,147 @@ class SubFilterViewModelTest {
 
         assertThat(viewModel.updateTagsAndSites.value).isEqualTo(null)
 
-        // we didn't call start so noone should have changed the value
-        assertThat(viewModel.currentSubFilter.value).isEqualTo(null)
+        assertThat(viewModel.currentSubFilter.value).isInstanceOf(SiteAll::class.java)
+    }
+
+    @Test
+    fun `view model updates the tags and sites and asks to show the bottom sheet when filters button is tapped`() {
+        var updateTasks: EnumSet<ReaderUpdateLogic.UpdateTask>? = null
+        var uiState: BottomSheetUiState? = null
+
+        viewModel.updateTagsAndSites.observeForever { updateTasks = it.peekContent() }
+        viewModel.bottomSheetUiState.observeForever { uiState = it.peekContent() }
+
+        viewModel.onSubFiltersListButtonClicked()
+
+        assertThat(updateTasks).isEqualTo(EnumSet.of(
+                UpdateTask.TAGS,
+                UpdateTask.FOLLOWED_BLOGS
+        ))
+
+        assertThat(uiState).isInstanceOf(BottomSheetVisible::class.java)
+    }
+
+    @Test
+    fun `view model hides the bottom sheet when it is cancelled`() {
+        var uiState: BottomSheetUiState? = null
+
+        viewModel.bottomSheetUiState.observeForever { uiState = it.peekContent() }
+
+        viewModel.onBottomSheetCancelled()
+
+        assertThat(uiState).isInstanceOf(BottomSheetHidden::class.java)
+    }
+
+    @Test
+    fun `bottom sheet is hidden when a filter is tapped on`() {
+        var uiState: BottomSheetUiState? = null
+        val filter: SubfilterListItem = mock()
+
+        viewModel.setSubfilterFromTag(savedTag)
+
+        viewModel.bottomSheetUiState.observeForever { uiState = it.peekContent() }
+
+        viewModel.getCurrentSubfilterValue().onClickAction!!.invoke(filter)
+
+        assertThat(uiState).isInstanceOf(BottomSheetHidden::class.java)
+    }
+
+    @Test
+    fun `onSaveInstanceState puts expected parameters in the bundle`() {
+        val outState: Bundle = mock()
+        whenever(subfilterListItemMapper.toJson(any())).thenReturn("test-json")
+
+        viewModel.onSaveInstanceState(outState)
+
+        verify(outState, times(1)).putString(eq(Companion.ARG_CURRENT_SUBFILTER_JSON), any())
+        verify(outState, times(1)).putBoolean(eq(Companion.ARG_IS_FIRST_LOAD), any())
+    }
+
+    @Test
+    fun `view model start register event bus`() {
+        verify(eventBusWrapper, times(1)).register(viewModel)
+    }
+
+    @Test
+    fun `onSubfilterSelected requests newer posts`() {
```

Comment:
I noticed two tests - `onSubfilterSelected requests newer posts` and `onSubfilterReselected does not request newer posts`. Can we mention a condition for requesting/not requesting posts in the method name (if it's applicable here)?

Reply:
Agree it makes it more clear 👍 ! Changed in acf5350